tensorcircuit/templates/lattice.py

```diff
@@ -1228,98 +1464,66 @@ def _build_lattice(self, *args: Any, **kwargs: Any) -> None:
         pass
 
     def _build_neighbors(self, max_k: int = 1, **kwargs: Any) -> None:
-        """Calculates neighbors using a KDTree for efficiency.
-
-        This method uses a memory-efficient approach to identify neighbors without
-        initially computing the full N x N distance matrix. It leverages
-        `scipy.spatial.distance.pdist` to find unique distance shells and then
-        a `scipy.spatial.KDTree` for fast radius queries. This approach is
-        significantly more memory-efficient during the neighbor identification phase.
-
-        After the neighbors are identified, the full distance matrix is computed
-        from the pairwise distances and cached for potential future use.
+        """
+        Calculates neighbor relationships using a distance matrix.
 
-        :param max_k: The maximum number of neighbor shells to
-            calculate. Defaults to 1.
-        :type max_k: int, optional
-        :param tol: The numerical tolerance for distance
-            comparisons. Defaults to 1e-6.
-        :type tol: float, optional
+        This method leverages the generic `_build_neighbors_by_distance_matrix`
```

Comment:
how the performance drop without KDTree? I think neighbor building can be on non differentiable, as they are discrete pairs and cannot be differentiated anyway? only the real valued distance matrix has to be differentiated.